modules/client/main.tf

```diff
@@ -0,0 +1,142 @@
+  template = "${local.client_startup_script}"
+
+  vars {
+    forseti_environment      = "${data.template_file.forseti_client_environment.rendered}"
+    forseti_repo_url         = "${var.forseti_repo_url}"
+    forseti_version          = "${var.forseti_version}"
+    forseti_home             = "${var.forseti_home}"
+    forseti_client_conf_path = "${local.client_conf_path}"
+    storage_bucket_name      = "${local.client_bucket_name}"
+  }
+}
+
+data "template_file" "forseti_client_environment" {
+  template = "${local.client_env_script}"
+
+  vars {
+    forseti_home             = "${var.forseti_home}"
+    forseti_client_conf_path = "${local.client_conf_path}"
+  }
+}
+
+data "template_file" "forseti_client_config" {
+  template = "${local.client_conf}"
+
+  vars {
+    forseti_server_ip = "${var.server_address}"
+  }
+}
+
+#-------------------#
+# Forseti client VM #
+#-------------------#
+resource "google_compute_instance" "forseti-client" {
+  name                      = "${local.client_name}"
+  zone                      = "${local.client_zone}"
+  project                   = "${var.project_id}"
+  machine_type              = "${var.client_type}"
+  allow_stopping_for_update = true
+
+  boot_disk {
+    initialize_params {
+      image = "${var.client_boot_image}"
+    }
+  }
+
+  network_interface {
+    subnetwork_project = "${var.network_project}"
+    subnetwork         = "${var.subnetwork}"
+
+    access_config {}
+  }
+
+  metadata_startup_script = "${data.template_file.forseti_client_startup_script.rendered}"
+
+  service_account {
+    email  = "${google_service_account.forseti_client.email}"
+    scopes = ["cloud-platform"]
+  }
+
+  depends_on = ["null_resource.services-dependency"]
+}
+
+#----------------------#
+# Forseti client roles #
+#----------------------#
+resource "google_project_iam_member" "client_roles" {
+  count   = "${length(local.client_project_roles)}"
+  role    = "${local.client_project_roles[count.index]}"
+  project = "${var.project_id}"
+  member  = "serviceAccount:${google_service_account.forseti_client.email}"
+}
+
+#-------------------------#
+# Forseti service Account #
+#-------------------------#
+resource "google_service_account" "forseti_client" {
+  account_id   = "${local.client_sa_name}"
+  project      = "${var.project_id}"
+  display_name = "Forseti Server Service Account"
+}
+
+#------------------------#
+# Forseti storage bucket #
+#------------------------#
+resource "google_storage_bucket" "client_config" {
+  name          = "${local.client_bucket_name}"
+  location      = "${var.storage_bucket_location}"
+  project       = "${var.project_id}"
+  force_destroy = "true"
+}
+
+resource "google_storage_bucket_object" "forseti_client_config" {
+  name       = "configs/forseti_conf_client.yaml"
+  bucket     = "${google_storage_bucket.client_config.name}"
+  content    = "${data.template_file.forseti_client_config.rendered}"
+  depends_on = ["google_compute_instance.forseti-client"]
```

Comment:
What is the purpose of this explicit dependency?

Reply:
Fixed in 65b1392.